packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,151 @@
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: intervalMinutes,
+          canaryPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + percentage+ 'Percent' + intervalMinutes + 'Minutes';
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: intervalMinutes,
+          linearPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaLinear' + percentage + 'Percent' + intervalMinutes + 'Minutes';
+    }
+
+    // What the config is called in the AWS console
+    // We tack the id on so that, if the user has multiple configs with identical settings,
+    // deleting one will not affect the other
+    const resourceName = id + customDeploymentConfigName;
+    this.deploymentConfigName = resourceName;
+    this.deploymentConfigArn = arnForDeploymentConfig(resourceName);
+
+    // The AWS Custom Resource that calls CodeDeploy to create and delete the resource
+    new AwsCustomResource(this, 'DeploymentConfig', {
+      onCreate: { // Run on creation only, to make the resource
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        // This `resourceName` is the initial physical ID of the config
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onUpdate: { // Run on stack update
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+          computePlatform: 'Lambda',
+          trafficRoutingConfig: routingConfig,
+        },
+        // If `resourceName` is different from the last stack update (or creation),
+        // the old config gets deleted and the new one is created
+        physicalResourceId: PhysicalResourceId.of(resourceName),
+      },
+      onDelete: { // Run on deletion, or on resource replacement
+        service: 'CodeDeploy',
+        action: 'deleteDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
+        },
+      },
+      // Least permissions, only have permission to create or delete this exact config
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: [this.deploymentConfigArn],
+      }),
+    });
+  }
+
+  private validateParameters(props: CustomLambdaDeploymentConfigProps): void {
+    if (!Number.isInteger(props.percentage)) {
+      throw new Error('Invalid deployment config percentage "' + props.percentage.toString()
+        + '", only integer values are acceptable.');
+    }
+    if (!Number.isInteger(props.interval.toMinutes())) {
```

Comment:
This is not a concern, this will always be a number... You can remove this validation

(Is there any validation  for minimum-maximum values here instead, that would be interesting...?)

Reply:
I'm checking whether the number is a pure integer, or if it is a floating point value. You cannot specify decimals in the config, CodeDeploy will throw an error if you try.

I tried specifying "1.5" for example, and it said "the nearest valid values are 1 and 2". I included this in there to prevent users from running into this at deployment time.

Reply:
But how can you construct a `Duration` that returns `toMinutes()` as a non-integer...?

I feel like checking for a min and max value will cover this case anyway.

Reply:
Just checked the `Duration` code, didn't realize it turns them into integers. I can remove this.